### PR TITLE
Add tombstone parsing, and force facts to be in a namespace (even tho…

### PR DESCRIPTION
**Pull request by HuwCampbell**
…ugh it's not used yet).

Evaluators currently don't know how to deal with tombstones either.

### CODE REVIEW

**Comment by tranma** [comment]
should evaluators handled tombstones as described in https://github.com/ambiata/icicle/blob/master/doc/design/tombstones.md?

**Comment by jystic** [comment]
@tranma yep

File: src/Icicle/Storage/Dictionary/TextV1.hs [hunk 1]
```diff
@@ -14,7 +14,8 @@ import           P hiding (concat, intercalate)
 import           Data.Attoparsec.Text
 
 import           Data.Either.Combinators
-import           Data.Text hiding (takeWhile)
+import           Data.Text hiding (takeWhile, singleton)
+import           Data.Set (singleton)
```
Comment (jystic) [inline src/Icicle/Storage/Dictionary/TextV1.hs]: Should really use qualified imports here

File: test/Icicle/Test/Encoding.hs [hunk 2]
```diff
@@ -7,11 +7,12 @@ module Icicle.Test.Encoding where
 import           Icicle.Test.Arbitrary
 import           Icicle.Encoding
 
-import           P
+import           P hiding (empty)
 
 import           System.IO
 
 import           Test.QuickCheck
+import           Data.Set (empty)
```
Comment (jystic) [inline test/Icicle/Test/Encoding.hs]: Qualified imports

**Comment by jystic** [comment]
:tangerine: looks good to me

might be good to get @amosr to have a look though (i think he might be in today), I'm not sure what the implications of adding `VTombstone` to `BaseValue` are

**Comment by amosr** [comment]
ok! :banana:

**Pull request merged**